src/compiler/checker.ts

```diff
@@ -11882,8 +11882,18 @@ namespace ts {
                 if (symbol.flags & SymbolFlags.Property &&
                     (expr.kind === SyntaxKind.PropertyAccessExpression || expr.kind === SyntaxKind.ElementAccessExpression) &&
                     (expr as PropertyAccessExpression | ElementAccessExpression).expression.kind === SyntaxKind.ThisKeyword) {
-                    const func = getContainingFunction(expr);
-                    return !(func && func.kind === SyntaxKind.Constructor && func.parent === symbol.valueDeclaration.parent);
+                    return !isInConstructor(getContainingFunction(expr));
+                    function isInConstructor(func: FunctionLikeDeclaration) {
+                        if (!func)

```

Comment:
nit: I think this "if" clause and below should be merged as it willl be easier to read

```typescript
if (!func || func.kind !== SyntaxKind.Constructor) {
    return false;
}
```